parsers/test/test_ERCOT.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+"""Tests for US_ERCOT.py"""
+
+import json
+import logging
+import unittest
+from datetime import datetime
+from unittest.mock import patch
+from zoneinfo import ZoneInfo
+
+from testfixtures import LogCapture
+
+from parsers import US_ERCOT
+
+
+class TestUSERCOT(unittest.TestCase):
+    """Patches in a fake response from the data source to allow repeatable testing."""
+
+    def test_fetch_production(self):
+        filename = "parsers/test/mocks/ERCOT.json"
+        with open(filename) as f:
+            fake_data = json.load(f)
+
+        with LogCapture(), patch("parsers.US_ERCOT.get_data") as gjd:
+            gjd.return_value = fake_data
+            data = US_ERCOT.fetch_production(logger=logging.getLogger("test"))
+
+        with self.subTest():
+            self.assertIsNotNone(data)
+        with self.subTest():
+            self.assertEqual(data[0]["production"]["coal"], 5067.958333)
+        with self.subTest():
+            expected_dt = datetime(2024, 11, 24, 8, 0, tzinfo=ZoneInfo("US/Central"))
+            self.assertEqual(data[-1]["datetime"], expected_dt)
+        with self.subTest():
+            self.assertEqual(data[-1]["source"], "ercot.com")
+        with self.subTest():
+            self.assertEqual(data[-1]["zoneKey"], "US-TEX-ERCO")
+        with self.subTest():
+            self.assertIsInstance(data[-1]["storage"], dict)
+
+    def test_fetch_consumption(self):
+        filename = "parsers/test/mocks/ERCOT_demand.json"
+        with open(filename) as f:
+            fake_data = json.load(f)
+
+        with LogCapture(), patch("parsers.US_ERCOT.get_data") as gjd:
+            gjd.return_value = fake_data
+            data = US_ERCOT.fetch_consumption(logger=logging.getLogger("test"))
+
+        with self.subTest():
+            self.assertIsNotNone(data)
+        with self.subTest():
+            self.assertEqual(data[0]["consumption"], 42015.44)
+        with self.subTest():
+            expected_dt = datetime(2024, 11, 24, 0, 0, tzinfo=ZoneInfo("US/Central"))
+            self.assertEqual(data[0]["datetime"], expected_dt)
+        with self.subTest():
+            self.assertEqual(data[0]["source"], "ercot.com")
+        with self.subTest():
+            self.assertEqual(data[0]["zoneKey"], "US-TEX-ERCO")
```

Comment:
Do you think we can add a snapshot test as well, so if we change things in the future we can see exactly what has changed?

Reply:
Felix said that he does this think testing in parsers is that important, so I'll let the two of you decide if it is worth the time or not.

Reply:
Agreed, I'll see if I can add a snapshot test when I get back home. Should be fairly quick since most of the test infra needed is already there.

A snapshot just makes it easy to spot subtle changes in the values of we ever change something this depend on, like the EIA parser or the parser classes.